packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -282,7 +282,14 @@ class IOSDevice extends Device {
       // TODO(chinmaygarde): Use mainPath, route.
       printTrace('Building ${package.name} for $id');
 
-      final String cpuArchitecture = await iMobileDevice.getInfoForDevice(id, 'CPUArchitecture');
+      String cpuArchitecture;
+
+      try {
+        cpuArchitecture = await iMobileDevice.getInfoForDevice(id, 'CPUArchitecture');
+      } on IOSDeviceNotFoundError {
+        return LaunchResult.failed();
```

Comment:
This should probably do a `printError` with the exception in it so the user knows what happened.

Reply:
Oh yeah, since we're swallowing the exception.

Reply:
done, PTAL